pkg/operator/apis/monitoring/v1/http_types.go

```diff
@@ -20,40 +20,142 @@ import (
 	"net/url"
 
 	"github.com/prometheus/common/config"
+	"github.com/prometheus/prometheus/secrets"
+	"github.com/prometheus/prometheus/secrets/kubernetes"
 )
 
-// Auth sets the `Authorization` header on every scrape request.
-//
-// Currently the credentials are not configurable and always empty.
+// SecretSelector represents a reference to a secret from the given provider
+// e.g. Kubernetes Secret. Only one provider can be used at the same time.
+type SecretSelector struct {
+	// KubernetesSecret represents reference to a given key from certain Kubernetes Secret
+	// in a given namespace.
+	// +optional
+	KubernetesSecret *KubernetesSecretKeySelector `json:"kubernetesSecret,omitempty"`
+}
+
+func (s *SecretSelector) toPrometheusSecretRef(m PodMonitoringCRD, pool PrometheusSecretConfigs) (string, error) {
+	if s == nil {
+		return "", nil
+	}
+
+	if s.KubernetesSecret == nil {
+		return "", nil
+	}
+
+	return s.KubernetesSecret.toPrometheusSecretRef(m, pool)
+}
+
+// KubernetesSecretKeySelector represents selector for Kubernetes secret.
+// It's similar to k8s.io/api/core/v1.SecretKeySelector, but
+// allows cross namespace selections.
+type KubernetesSecretKeySelector struct {
+	// Name of the secret to select from.
+	Name string `json:"name"`
```

Comment:
I learn towards our own struct because:
1. `LocalObjectReference` only has 1 field. Not much value.
2. `LocalObjectReference` documentation is confusing. What type of "referent"?
3. We always inline it. Extra layer of indirection for no reason.
4. `LocalObjectReference` may add additional fields that we'd have to support or decide to replace this object. Hypothetically this type should be a "local" version of [`ObjectReference`](https://pkg.go.dev/k8s.io/api/core/v1?utm_source=godoc#ObjectReference), so I can envision them adding these fields. See [official discussions]( https://github.com/kubernetes-retired/service-catalog/issues/1411).
5. `corev1.SecretKeySelector` has an `optional` field which we'll never use. We're not using it in other places -- does this mean our implementation is incorrect/non-conforming?

Reply:
Sounds like you have strong opinions so SGTM